### PR TITLE
Simplify how we generate special types

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
This has net benefits to our codegen featuers as we can avoid generating fully qualified names like `global::System.Int32` which we're just going to have to reanalyze later to determine if we can simplify it to `int`.  now we just generate the reduced type to begin with which we know will always bind correctly as per the rules of hte language.

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ITypeSymbolExtensions.cs [hunk 1]
```diff
@@ -96,6 +97,39 @@ internal static partial class ITypeSymbolExtensions
             return SyntaxFactory.RefType(refKeyword, readOnlyKeyword, underlyingType);
         }
 
+        public static bool TryCreateTypeExpression(SpecialType specialType, [NotNullWhen(true)] out PredefinedTypeSyntax? node)
```
Comment (genlu) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ITypeSymbolExtensions.cs:100]: Consider making this an extension method, which is more discoverable via completion.

**Review by genlu — APPROVED** [review]
:shipit:

**Review by genlu — COMMENTED** [review]
Actually, the test failure is related, please fix

**Comment by Youssef1313** [comment]
Does this fix the test flakiness that appeared recently?

**Comment by CyrusNajmabadi** [comment]
Closing out for now.  @sharwell makes a good point that this can likely interfere with the user option to use named types over predefined types.  That's solvable, but likely risky.  We can return to this another time.  Note that thsi does mean that some of our features likely get this option wrong *if* they are using SyntaxGenerator to synthesize types.

**Pull request closed without merging**